src/test/groovy/org/gradle/profiler/ChromeTraceIntegrationTest.groovy

```diff
@@ -55,9 +59,38 @@ class ChromeTraceIntegrationTest extends AbstractProfilerIntegrationTest {
             run("--project-dir", projectDir.absolutePath, "--output-dir", outputDir.absolutePath, "--gradle-version", versionUnderTest, "--profile", "chrome-trace", "--no-daemon", "assemble")
 
         then:
-        new File(outputDir, "${versionUnderTest}-trace.json").isFile()
+        new File(outputDir, "${versionUnderTest}-trace/${versionUnderTest}-warm-up-build-1-invocation-1-trace.json").isFile()
+        new File(outputDir, "${versionUnderTest}-trace/${versionUnderTest}-measured-build-1-invocation-1-trace.json").isFile()
 
         where:
         versionUnderTest << supportedGradleVersionsForChromeTrace
     }
+
+    @Requires({ StudioFinder.findStudioHome() })
+    def "profiles build to produce chrome trace output for builds with multiple gradle invocations"() {
+        given:
+        def studioHome = StudioFinder.findStudioHome()
+        new File(projectDir, "buildSrc").mkdirs()
+        new File(projectDir, "buildSrc/gradle.build").createNewFile()
+        def scenarioFile = file("performance.scenarios") << """
+            scenario {
+                android-studio-sync {}
+            }
+        """
+
+        when:
+        new Main().run("--project-dir", projectDir.absolutePath, "--output-dir", outputDir.absolutePath,
+            "--gradle-version", latestSupportedGradleVersion,
+            "--profile", "chrome-trace",
+            "--scenario-file", scenarioFile.absolutePath,
+            "--studio-install-dir", studioHome.absolutePath,
+            "--warmups", "1",
+            "--iterations", "1")
+
+        then:
+        new File(outputDir, "scenario-${latestSupportedGradleVersion}-trace/scenario-${latestSupportedGradleVersion}-warm-up-build-1-invocation-1-trace.json").isFile()
```

Comment:
Have we also been generating trace files for warmup builds? I somewhat was expecting that we'd only produce the traces for the actual iterations. Note that I am not sure what is better here.

Reply:
It looks like we do yes. Since it's really low overhead I think we can leave it as it is